### DISCUSSION TITLE
Wait element then send a custom search url

### DISCUSSION THREAD

**Discussion by Fred-Vatin** [hoothin/SearchJumper#179] [Q&A] [answered]
Hi,

I have a custom search engine such as
```
https://www.ygg.re/engine/search?name=%s&description=&file=&uploader=&category=2145&sub_category=all&do=search&order=desc&sort=seed
```

It works only if I’m previously logged in otherwise it opens `https://www.ygg.re/auth/login` to login.

This will work to wait until I’m logged in:
```
https://www.ygg.re/#p{wait(form.search input)&form.search input=%s&sleep(400)&click(form.search button)}
```

Nice but I don’t want to process the default search but the one given by
```
https://www.ygg.re/engine/search?name=%s&description=&file=&uploader=&category=2145&sub_category=all&do=search&order=desc&sort=seed
```
because it uses filters and custom sort.

Any idea how I could mix
```
https://www.ygg.re/#p{wait(form.search input)}
```
that valid I’m logged in, then send the custom search url
```
https://www.ygg.re/engine/search?name=%s&description=&file=&uploader=&category=2145&sub_category=all&do=search&order=desc&sort=seed
```
???

**Comment by hoothin** [comment]
First, create an engine named 'xxx' with the content `https://www.ygg.re/engine/search?name=%s`
And then use a second engine to call it. like: `https://www.ygg.re/#p{wait(form.search input)&call(xxx)}`

**Reply by Fred-Vatin** [reply]
Brilliant. But the search term is not passed to the called engine.

### Setup
I created an engine with the URL I need using %s to use the term sent by the caller.

Name: `YGG vidéo bypass`
URL: `https://www.ygg.re/engine/search?name=%s.replace(/(\s*)&(\s*)/g, "$1and$2").replace(/.*demon slayer.*/ig, "demon slayer")&description=&file=&uploader=&category=2145&sub_category=all&do=search&order=desc&sort=seed`
I set it to open it in the current tab.

I created the caller with

Name: `YGG vidéo`
URL: `https://www.ygg.re/#p{wait(form.search input)&call(YGG vidéo bypass)}`

### Result
The logging page is bypassed then the called engine runs. But it opens the input box to ask my search term. Is the call function not supposed to send the search term so I can use `%s` in the called engine ?

So I tried to use `%S` instead. In this case, no input box but it sends this url to the browser:
```
https://www.ygg.re/engine/search?name=engine.replace(/(\s*)&(\s*)/g,%20%22$1and$2%22).replace(/.*demon%20slayer.*/ig,%20%22demon%20slayer%22)&description=&file=&uploader=&category=2145&sub_category=all&do=search&order=desc&sort=seed
```
The bypass `YGG vidéo bypass` engine works if used directly but not as expected via call. How to get rid of the input box ?

**Reply by hoothin** [reply]
Are you currently accessing this engine within this website (ygg.re)?

**Reply by Fred-Vatin** [reply]
Not necessarily. It doesn’t change the result.

**Reply by hoothin** [reply]
That's weird. Could you try the latest version and see if the issue is resolved?
https://addons.mozilla.org/firefox/addon/searchjumper/

**Reply by Fred-Vatin** [reply]
I updated and restarted Firefox to be sure. Same results. It asks me an input.

**Reply by hoothin** [reply]
So I tested it again and didn't find any issues. The selected text was sent to the `call()` engine normally. Could you provide me with a complete operation video?

**Reply by hoothin** [reply, marked as answer]
Wait, I think I've found the problem.

**Reply by Fred-Vatin** [reply]
Last version fixes this issue. Thanks for your hard work.
